Copy edits to client session docs (#229)
seamapi/api-docs: docs/core-concepts/workspaces/client-session-tokens.md

---
description: >-

>   Use a client session token to restrict access to the devices owned by a specific user user.

Reviewer: to the devices that a specific user owns.

>   Use a client session token to restrict access to the devices owned by a specific user user.

Reviewer: ```suggestion
  Use a client session token to restrict access to the devices that a specific user owns.
```

---

# Client Session Tokens

> If you want to restrict your users' access to their own devices, use a Client Session Token.

Reviewer: client session token (lowercase)

> If you want to restrict your users' access to their own devices, use a Client Session Token.

Reviewer: ```suggestion
If you want to restrict your users' access to their own devices, use a client session token.
```

> Use Client Session Tokens with the Seam JavaScript SDK to authenticate and authorize users of you web application directly with Seam.

Reviewer: client session token (lowercase)
your web app

> Use Client Session Tokens with the Seam JavaScript SDK to authenticate and authorize users of you web application directly with Seam.

Reviewer: ```suggestion
Use client session tokens with the Seam JavaScript SDK to authenticate and authorize users of you web application directly with Seam.
```

> This eliminates the need for your user's browser to communicate with your back-end server.

Reviewer: Using client session tokens eliminates...

> This eliminates the need for your user's browser to communicate with your back-end server.

Reviewer: ```suggestion
Using client session tokens eliminates the need for your user's browser to communicate with your back-end server.
```

> Instead, requests are made directly from the client to the Seam API.

Reviewer: Instead, the client makes requests directly to the Seam API.

> Instead, requests are made directly from the client to the Seam API.

Reviewer: ```suggestion
Instead, the client makes requests directly to the Seam API.
```

{% hint style="info" %}

> If you are just getting started with Seam, check out Seam Components.

Reviewer: Need to add a hyperlink to the Seam Components topic.

> If you are just getting started with Seam, check out Seam Components.

Reviewer: ```suggestion
If you are just getting started with Seam, see [Seam Components](../../seam-components/overview).
```

> In only a few minutes and with just a few lines of code, you can start controlling devices from your web app without touching your backend.

Reviewer: without interacting with your back end

> In only a few minutes and with just a few lines of code, you can start controlling devices from your web app without touching your backend.

Reviewer: ```suggestion
In only a few minutes and with just a few lines of code, you can start controlling devices from your web application without interacting with your back end.
```

{% endhint %}

> <figure><img src="../../.gitbook/assets/client-session-diagram.png" alt="Use client session tokens to restrict users&#x27; access to only the devices that they own."><figcaption></figcaption></figure>

Reviewer: We need to reorg this section so that we don't have two diagrams right next to each other without any text in between them.

Author: I moved the main diagram to the client session page since it applies to all client sessions independently from the concept of using the "client-side token"

<figure><img src="../../.gitbook/assets/client-session-flow.png" alt="Client session tokens increase efficiency by enabling your user&#x27;s browser to communicate directly with the Seam back-end server."><figcaption></figcaption></figure>

{% hint style="warning" %}

> A Client Session Token may only be used in a browser context via the Seam JavaScript SDK.

Reviewer: You can only use a client session token in a browser context through the Seam JavaScript SDK.

> A Client Session Token may only be used in a browser context via the Seam JavaScript SDK.

Reviewer: ```suggestion
You can only use a client session token in a browser context through the Seam JavaScript SDK.
```

> Do not use a Client Session Token to make requests from your backend, instead use an [API key](api-keys.md).

Reviewer: client session token (lowercase) ... back end. Instead, use ...

> Do not use a Client Session Token to make requests from your backend, instead use an [API key](api-keys.md).

Reviewer: ```suggestion
Do not use a client session token to make requests from your back end. Instead, use an [API key](api-keys.md).
```

> An API Key has unlimited permission to manage all devices and other Seam API resources within a workspace.

Reviewer: key (lowercase)

> Never use an API Key in the browser or expose it to your users.

Reviewer: key (lowercase)

(We should also add this warning ["Never use an API Key in the browser or expose it to your users."] to the API keys topic!)

> An API Key has unlimited permission to manage all devices and other Seam API resources within a workspace.
> Never use an API Key in the browser or expose it to your users.

Reviewer: ```suggestion
An API key has unlimited permission to manage all devices and other Seam API resources within a workspace.
Never use an API key in the browser or expose it to your users.
```

{% endhint %}

> A Client Session Token can either be issued from your backend to the client, or managed entirely from the client using [Seam Components with a Publishable Key](../../seam-components/overview/get-started-with-client-side-components.md).

Reviewer: You can issue a client session token in either of the following two ways:
- Issue from your back end [or server] to the client.
- Manage entirely from the client using [Seam Components with a publishable key](../../seam-components/overview/get-started-with-client-side-components.md).

> A Client Session Token can either be issued from your backend to the client, or managed entirely from the client using [Seam Components with a Publishable Key](../../seam-components/overview/get-started-with-client-side-components.md).

Reviewer: ```suggestion
You can issue a client session token in either of the following two ways:

- Issue from your back end [or server] to the client.
- Manage entirely from the client using [Seam Components with a publishable key](../../seam-components/overview/get-started-with-client-side-components.md).
```

Author: ```suggestion
You authorize the Seam SDK client session token:

- Create or retrieve a client session token from your back end and pass it to your front end to use with the Seam JavaScript SDK or Seam Components.
- Manage client sessions entirely on the front end by using a [publishable key with the JavaScript SDK or Seam Components](../../seam-components/overview/get-started-with-client-side-components.md).
```

> To generate a client session token, from your backend, use the Seam API to create a client session that includes [your own internal user ID for the user](../../seam-components/overview/get-started-with-client-side-components.md#3-select-a-user-identifier-key), along with the [connected accounts](../../api-clients/connected-accounts/) associated with this user. The new client session contains a token that you can pass to the Seam JavaScript SDK running in the browser.

Reviewer: back end [or server]

> To generate a client session token, from your backend, use the Seam API to create a client session that includes [your own internal user ID for the user](../../seam-components/overview/get-started-with-client-side-components.md#3-select-a-user-identifier-key), along with the [connected accounts](../../api-clients/connected-accounts/) associated with this user. The new client session contains a token that you can pass to the Seam JavaScript SDK running in the browser.

Reviewer: ```suggestion
To generate a client session token, from your back end, use the Seam API to create a client session that includes [your own internal user ID for the user](../../seam-components/overview/get-started-with-client-side-components.md#3-select-a-user-identifier-key), along with the [connected accounts](../../api-clients/connected-accounts/) associated with this user. The new client session contains a token that you can pass to the Seam JavaScript SDK running in the browser.
```

***

## Create a Client Session for an Existing User and Retrieve the Session Token

To [create a client session](../../api-clients/client-sessions/create-a-client-session.md) for an existing user with connected accounts in your workspace, include the IDs of the user's connected accounts (`connected_account_ids`) and [provide your own unique internal user ID (`user_identifier_key`)](../../seam-components/overview/get-started-with-client-side-components.md#3-select-a-user-identifier-key).

> Then, use the generated client session token to retrieve and manage the resources authorized via the connected accounts associated with this client session.

Reviewer: authorized through

> Then, use the generated client session token to retrieve and manage the resources authorized via the connected accounts associated with this client session.

Reviewer: ```suggestion
Then, use the generated client session token to retrieve and manage the resources authorized through the connected accounts associated with this client session.
```

{% tabs %}
{% tab title="JavaScript" %}
**Request:**

```javascript
const client_session = await seam.clientSessions.create({
  user_identifier_key: "internal-user-id-1",
  connected_account_ids: [
    "3ea0b67f-649f-4131-bfe3-f2035e77a3f9",
    "6e1cad57-b244-40ca-b4f3-30a46c8000d4",

>   ]

Reviewer: Same comments as for other code snippets.

**Request:**

```javascript

> await seam.clientSessions.list()

Reviewer: Same comments here too.

**Request:**

```javascript
const clientSession = await seam.clientSessions.get({
  client_session_id: "c75d4330-ae01-4dfd-b6c5-f3e94e0d8168",

> }))

Reviewer: Same comments here too.

**Request:**

```javascript
await seam.clientSessions.delete({
  client_session_id: "639eb8af-19a9-4ae5-817c-6ad08fd078e7",

> })

Reviewer: Same comments here too.

## Next Steps

> - Learn about using [Client Session Tokens with Seam Components](../../seam-components/overview).

Reviewer: ```suggestion
- Learn about using [client session tokens with Seam Components](../../seam-components/overview).
```

> - Learn more about [Connect Webviews](../connect-webviews/).

Reviewer: The paragraph style is to keep consistent with other guide topics. Happy to change this section to a bulleted list, but, if we do that, we should change all other next steps sections accordingly.

Author: Follow sections here. Can switch to paragraph style, but bulleted list seems a lot easier to read and pick out a link or guide I think it relevant to my next use goal.

- https://github.com/seamapi/api-docs/pull/229/files?diff=unified&w=0#r1439914875
- https://github.com/seamapi/api-docs/pull/229/files?diff=unified&w=0#r1439914875
- https://github.com/seamapi/api-docs/pull/229/files?diff=unified&w=0#r1439914875
